src/Views/Edit.php

```diff
@@ -0,0 +1,161 @@
+                    ->field()
+                    ->select(
+                        $this->translator->translate('aspect_ratio'),
+                        $this->ratio_option,
+                        $this->translator->translate('aspect_ratio_info')
+                    )
+                    ->withValue((string) $closest_ratio)
+                    ->withRequired(true),
+                \ilOpencastPageComponentPluginGUI::PROP_WIDTH => $this->ui_factory
+                    ->input()
+                    ->field()
+                    ->numeric(
+                        $this->translator->translate('max_width'),
+                        $this->translator->translate('max_width_info')
+                    )
+                    ->withValue($this->properties[\ilOpencastPageComponentPluginGUI::PROP_WIDTH] ?? null),
+                \ilOpencastPageComponentPluginGUI::PROP_AS_LINK => $this->ui_factory
+                    ->input()
+                    ->field()
+                    ->checkbox(
+                        $this->translator->translate('link_thumbnail'),
+                        $this->translator->translate('link_thumbnail_info')
+                    )
+                    ->withValue((bool) ($this->properties[\ilOpencastPageComponentPluginGUI::PROP_AS_LINK] ?? false)),
+            ]
+        );
+    }
+
+    public function get(): Component|array
+    {
+        return [
+            $this->integration->events()->asItemFromEventId(
+                $this->properties[\ilOpencastPageComponentPluginGUI::PROP_EVENT_ID],
+                $this->ui_factory->button()->standard(
+                    $this->translator->translate('event_select_another'),
+                    (string) $this->replace_uri,
+                ),
+                null,
+                $this->translator->translate('event_currently_selected'),
+            ),
+            $this->ui_factory->panel()->standard(
+                $this->translator->translate('event_properties'),
+                $this->getPropertiesForm()
+            )
+        ];
+    }
+
+    public function getUpdatedProperties(RequestInterface $request): ?array
+    {
+        return $this->getPropertiesForm()->withRequest($request)->getData();
+    }
+
+    protected function determineRatio(): float
+    {
+        if (!empty($current = $this->properties[\ilOpencastPageComponentPluginGUI::PROP_ASPECT_RATIO] ?? null)) {
+            return (float) $current;
+        }
+
+        $event = $this->event_repository->find($this->properties[\ilOpencastPageComponentPluginGUI::PROP_EVENT_ID]);
+        $event->publications()->getThumbnailUrl();
+        // try to determine aspect ratio of the thumbnail
+        // get content of the thumbnail
+        $thumbnail = file_get_contents($event->publications()->getThumbnailUrl());
+        // get image size
+        $size = getimagesizefromstring($thumbnail);
+        // calculate aspect ratio
+        $aspect_ratio = $size[0] / $size[1];
```

Comment:
how about casting the whole division to float here?
```suggestion
        $aspect_ratio = (float) ($size[0] / $size[1]);
```

Reply:
What about here? @chfsx